iothub_client/src/iothub_client_core_ll.c

```diff
@@ -2281,6 +2281,24 @@ IOTHUB_CLIENT_RESULT IoTHubClientCore_LL_SetOption(IOTHUB_CLIENT_CORE_LL_HANDLE
             result = IOTHUB_CLIENT_ERROR;
 #endif /*DONT_USE_UPLOADTOBLOB*/
         }
+        else if (strcmp(optionName, OPTION_SAS_TOKEN_LIFETIME) == 0)
+        {
+            if (IoTHubClient_Auth_Set_SasToken_Expiry(handleData->authorization_module, *(size_t*)value) != 0)
+            {
+                LogError("Failed setting the Token Expiry time");
+                result = IOTHUB_CLIENT_ERROR;
+            }
+            else
+            {
+                result = IOTHUB_CLIENT_OK;
+            }
+        }
+        else if (strcmp(optionName, OPTION_SAS_TOKEN_REFRESH_TIME) == 0)
+        {
+            // This options is no longer needed, but may be updated in the future
+            // if this becomes necessary
+            result = IOTHUB_CLIENT_OK;
+        }
```

Comment:
Also this is needed to honor the intra-major-version contract we have with customers. Aka, so we don't break the API.

Reply:
Changed this to do the same as Sas Token Lifetime

---
In reply to: [251145567](https://github.com/Azure/azure-iot-sdk-c/pull/825#discussion_r251145567) [](ancestors = 251145567)